src/client/components/pages/RoomAdmin/RoomAdmin.tsx

```diff
@@ -147,6 +221,22 @@ const RoomAdmin: FunctionComponent = (): ReactElement => {
                   });
                 }}
               />
+              {editRoomModalData.room && (
```

Comment:
We want to avoid conditionally rendering modals like this, since the underlying `Modal` is already handling the show/hide behavior, and adding and removing child elements from the page-level component can trigger a full re-render. 

I know that sometimes causes problems in the underlying modal if `editRoomModalData.room` is undefined -- usually that means needing to add conditionals inside the modal to make sure we have fallbacks for those values.

Reply:
Thanks for explaining this! I removed the conditional rendering and added in optional chaining so that we cannot potentially trying to access properties of an undefined value. (commit [4ad1d77](https://github.com/seas-computing/course-planner/pull/581/commits/4ad1d77a066a039079968ec5c3be112551c4872b))